lib/documents.js

```diff
@@ -2716,6 +2716,253 @@ function urisOutputTransform(headers, data) {
+  let connectionParams = this.client.getConnectionParams();
+  let requestOptions = mlutil.copyProperties(connectionParams);
+
+  requestOptions.method = 'GET';
+  requestOptions.headers = {
+    'Accept':       'application/json',
+  };
+
+  requestOptions.path = mlutil.databaseParam(connectionParams, path);
+  let operation = new Operation(
+      'read forestInfo', this.client, requestOptions, 'empty', 'single'
+  );
+
+  let jobState = {
+    docInstance : this,
+    requesterCount: 0,
+    docsReadSuccessfully:0,
+    docsFailedToBeRead: 0,
+    jobOptions: (jobOptions)? mlutil.copyProperties(jobOptions):{}
+  };
+
+  let inputStream = null;
+  if(jobState.jobOptions.inputKind){
+    if (!new Set(['string', 'array']).has(jobState.jobOptions.inputKind)) {
+      throw new Error('Invalid value for inputKind. Value must be array or string.');
+    }
+    inputStream = jobState.jobOptions.inputKind === 'array'?
+        new stream.PassThrough({objectMode: true}): new stream.PassThrough();
+  } else {
+    inputStream = new stream.PassThrough({objectMode: true});
+  }
+ jobState.inputStream = inputStream;
+
+  let outputStream = null;
+  if(jobState.jobOptions.outputStreamType){
+    if (!new Set(['chunked', 'object']).has(jobState.jobOptions.outputStreamType)) {
+      throw new Error('Invalid value for outputStreamType. Value must be chunked or object.');
+    }
+    outputStream = jobState.jobOptions.outputStreamType === 'object'?
+        new stream.PassThrough({objectMode: true}): new stream.PassThrough();
+  } else {
+    outputStream = new stream.PassThrough({objectMode: true});
+  }
+  jobState.outputStream = outputStream;
+
+  let duplexStream = new stream.Duplex({
+    read(size){
+      return outputStream.read(size);
+    },
+    write(chunk, encoding, callback) {
+      inputStream.write(chunk, encoding, callback);
+    }
+  });
+
+  jobState.duplexStream = duplexStream;
+
+  if(jobState.jobOptions.concurrentRequests) {
+    const concurrentRequests = jobState.jobOptions.concurrentRequests;
+    if (concurrentRequests.multipleOf) {
+      if (!new Set(['forests', 'hosts']).has(String(concurrentRequests.multipleOf))) {
+        throw new Error('Invalid value for multipleOf. Value must be forests or hosts.');
+      }
+    } else {
+      concurrentRequests.multipleOf = 'forests';
+    }
+
+    if (concurrentRequests.multiplier && concurrentRequests.multiplier <= 0) {
+      throw new Error('concurrentRequests.multiplier cannot be less than one');
+    } else {
+      concurrentRequests.multiplier = 8;
+    }
+    jobState.jobOptions.concurrentRequests = concurrentRequests;
+  }
+  else {
+    jobState.jobOptions.concurrentRequests = {multipleOf:'forests', multiplier:8};
+  }
+  if(jobState.jobOptions.batchSize){
+    if(jobState.jobOptions.inputKind === 'array'){
+      throw new Error('batchSize not expected when inputKind is array.');
+    }
+    if(jobState.jobOptions.batchSize <=0){
+      jobState.jobOptions.batchSize = 250;
```

Comment:
Would it be better to throw an error if the request supplies an invalid batch size so the application developer can fix the input?